### ISSUE TITLE
Remove jsnext:main support from repo

### ISSUE THREAD

**Issue by shellscape** [rollup/plugins#83]
- Rollup Plugin Name: `node-resolve`, all others
- Rollup Plugin Version: n / a

### Expected Behavior / Situation

n / a 

### Actual Behavior / Situation

After @TrySound mentioned that `jsnext:main` is deprecated, I did some digging and asking around. Rollup was really the only bundler that embraced it fully. As mentioned by Rich Harris [here](https://twitter.com/Rich_Harris/status/1203299929285971968), `jsnext:main` has been superseded by `module`, and there are numerous sources that consider it deprecated as a result ([for example](https://github.com/stereobooster/package.json#jsnextmain)).

Basically it never made it out of discussion, and Rollup supported it out of necessity at the time it was implemented.

### Modification Proposal

So all that goes to say is that we have sufficient evidence to remove support from `@rollup/node-resolve` and to remove that field from any plugins in the repo.

**Comment by shellscape** [comment]
#85 took care of most of the work. Going to leave this open as several modules have to get a version bump and get published.
